tests/unit/test_http_client.py

```diff
@@ -223,3 +223,36 @@ def simple_json_callback(resp):
     assert exp_resp == resp
     assert 1 == mock_set_valid_token_called
     assert 2 == len(caplog.records)
+
+
+#####
+# Tests & fixtures the GPagintorMixin
+#####
+@pytest.fixture
+def simple_paging_client(mocker, get_gce_client):
+    class TestClient(http_client.GPaginatorMixin):
+
+        async def get_json(self, url, params=None):
+            return {'data': 'data', 'nextPageToken': 'token'}
+
+    return TestClient()
+
+
+@pytest.mark.asyncio
+@pytest.mark.parametrize('max_pages', [1, 2])
+async def test_get_all_pages(mocker, simple_paging_client, max_pages):
+    number_of_calls = 0
+
+    async def get_json(url, params=None):
+        nonlocal number_of_calls
+        if number_of_calls < (max_pages - 1):
+            number_of_calls += 1
+            return {'data': 'data', 'nextPageToken': 'token'}
+        else:
+            # last page doesn't include a nextPageToken
+            return {'data': 'final page'}
+    mocker.patch.object(simple_paging_client, 'get_json', get_json)
```

Comment:
I'm a little confused (and under-caffeinated) at why you're patching this when it's defined in the `TestClient` that `simple_paging_client` returns...

Reply:
Ah, good point, it was a left-over from when there were more tests for this code. Since there's just one test, I've pulled the fixture into the test itself.